src/array.jl

```diff
@@ -970,7 +970,9 @@ Base.repeat(a::CatArrOrSub{T, N};
 StructTypes.StructType(::Type{<:CategoricalVector}) = StructTypes.ArrayType()
 
 function StructTypes.construct(::Type{<:CategoricalArray}, A::Vector)
-    if eltype(A) >: Nothing
+    if eltype(A) == Any
```

Comment:
This can help the compiler:
```suggestion
    if eltype(A) === Any
```